cli/lib/kontena/cli/nodes/update_command.rb

```diff
@@ -3,18 +3,42 @@ class UpdateCommand < Kontena::Command
     include Kontena::Cli::Common
     include Kontena::Cli::GridOptions
 
-    parameter "NODE_ID", "Node id"
+    requires_current_master
+    requires_current_master_token
+    requires_current_grid
+
+    parameter "NAME", "Node name"
+
+    option ["--token"], "TOKEN", "Node token"
+    option "--generate-token", :flag, "Generate new node token"
+
     option ["-l", "--label"], "LABEL", "Node label", multivalued: true
+    option "--clear-labels", :flag, "Clear node labels"
+
+    def update_token
+      data = {}
+
+      data[:token] = self.token if self.token
+
+      spinner "Updating node #{name.colorize(:cyan)} token" do
+        client.put("nodes/#{current_grid}/#{name}/token", data)
+      end
+    end
 
     def execute
-      require_api_url
-      require_current_grid
-      token = require_token
+      self.update_token if self.token or self.generate_token?
 
       data = {}
-      data[:labels] = label_list if label_list
```

Comment:
The `if label_list` doesn't really work for multivalued options... an empty array is true.

This used to wipe out the node labels, now hurts because the `--label` is not the only option anymore.